Yoast/wordpress-seo
# [P2-445] Fix background link indexing

## Context


* Link indexing doesn't for the background job. Meaning that if you have a few links that are wrong they are not automatically fixed but require manually running the indexing process.

## Summary


This PR can be summarized in the following changelog entry:

* Improves automatic indexing to also index links.

## Relevant technical choices:

* Removes the check that the link_count is NULL to allow the indexing process to also fix links that are broken without impacting the link_count. This should be a rare edge-case, but worth fixing.

## Test instructions

This PR can be tested by following these steps:

* Change the `target_indexable_id` of a single row in the SEO links table to `NULL` where the target_post_id is not `NULL` or `0`.
* Navigate to any Yoast settings page.
* The ID should be filled again.

## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/QAK-2341
